retrieval/discovery/ec2.go

```diff
@@ -110,12 +118,34 @@ func (ed *EC2Discovery) refresh() (*config.TargetGroup, error) {
 				labels := model.LabelSet{
 					ec2LabelInstanceID: model.LabelValue(*inst.InstanceId),
 				}
-				if inst.PublicIpAddress != nil {
-					labels[ec2LabelPublicIP] = model.LabelValue(*inst.PublicIpAddress)
-				}
 				labels[ec2LabelPrivateIP] = model.LabelValue(*inst.PrivateIpAddress)
 				addr := fmt.Sprintf("%s:%d", *inst.PrivateIpAddress, ed.port)
 				labels[model.AddressLabel] = model.LabelValue(addr)
+
+				if inst.PublicIpAddress != nil {
+					labels[ec2LabelPublicIP] = model.LabelValue(*inst.PublicIpAddress)
+					labels[ec2LabelPublicDNS] = model.LabelValue(*inst.PublicDnsName)
+				}
+
+				labels[ec2LabelAZ] = model.LabelValue(*inst.Placement.AvailabilityZone)
+
+				if inst.VpcId != nil {
+					labels[ec2LabelVPCID] = model.LabelValue(*inst.VpcId)
+
+					subnetsMap := make(map[string]struct{})
+					for _, eni := range inst.NetworkInterfaces {
+						subnetsMap[*eni.SubnetId] = struct{}{}
+					}
+					subnets := []string{}
+					for k := range subnetsMap {
+						subnets = append(subnets, k)
+					}
+					labels[ec2LabelSubnetID] = model.LabelValue(
+						ed.subnetSeparator +
```

Comment:
That's how it is done for the Consul SD tags.
The comma before and after make regexing in relabel configurations significantly simpler.

The tags are spread over multiple labels because they are actually labels, i.e. key/value pairs, so having them in their own label allows actually adressing them. A list on the other hand could only have indexes suffixed to the label names. Here concatenating them with a separator is also simpler in relabel configurations.